tests/test_vips.py

```diff
@@ -38,136 +38,131 @@ def test_concurrency(self):
-                          image.shrink, xscale=0.0, yscale=1.0)
+                          VImageAdapter(image).shrink_affine, xscale=0.0, yscale=1.0)
         self.assertRaises(ValueError,
-                          image.shrink, xscale=2.0, yscale=1.0)
+                          VImageAdapter(image).shrink_affine, xscale=2.0, yscale=1.0)
         self.assertRaises(ValueError,
-                          image.shrink, xscale=1.0, yscale=0.0)
+                          VImageAdapter(image).shrink_affine, xscale=1.0, yscale=0.0)
         self.assertRaises(ValueError,
-                          image.shrink, xscale=1.0, yscale=2.0)
+                          VImageAdapter(image).shrink_affine, xscale=1.0, yscale=2.0)
 
     def test_tms_align(self):
-        image = VImage.new_rgba(width=16, height=16)
+        image = VImageAdapter.new_rgba(width=16, height=16)
 
         # Already aligned to integer offsets
-        result = image.tms_align(tile_width=16, tile_height=16,
+        result = VImageAdapter(image).tms_align(tile_width=16, tile_height=16,
                                  offset=XY(1, 1))
-        self.assertEqual(result.Xsize(), image.Xsize())
-        self.assertEqual(result.Ysize(), image.Ysize())
+        self.assertEqual(result.width, image.width)
+        self.assertEqual(result.height, image.height)
 
         # Spanning by half tiles in both X and Y directions
-        result = image.tms_align(tile_width=16, tile_height=16,
+        result = VImageAdapter(image).tms_align(tile_width=16, tile_height=16,
                                  offset=XY(1.5, 1.5))
-        self.assertEqual(result.Xsize(), image.Xsize() * 2)
-        self.assertEqual(result.Ysize(), image.Ysize() * 2)
+        self.assertEqual(result.width, image.width * 2)
+        self.assertEqual(result.height, image.height * 2)
 
         # Image is quarter tile
-        result = image.tms_align(tile_width=32, tile_height=32,
+        result = VImageAdapter(image).tms_align(tile_width=32, tile_height=32,
                                  offset=XY(1, 1))
-        self.assertEqual(result.Xsize(), image.Xsize() * 2)
-        self.assertEqual(result.Ysize(), image.Ysize() * 2)
-
-    def test_write_to_memory(self):
-        image = VImage.new_rgba(width=16, height=16)
-        image2 = image.write_to_memory()
-        self.assertEqual(image.tobuffer(), image2.tobuffer())
-
-    def test_write_to_tempfile(self):
-        image = VImage.new_rgba(width=16, height=16)
-        image2 = image.write_to_tempfile()
-        tempfile = image2._buf
-        self.assertEqual(image.tobuffer(), image2.tobuffer())
-
-        # Does the file delete itself?
-        self.assertTrue(os.path.exists(tempfile.name))
-        tempfile.close()
-        self.assertFalse(os.path.exists(tempfile.name))
+        self.assertEqual(result.width, image.width * 2)
+        self.assertEqual(result.height, image.height * 2)
+
+    # def test_write_to_memory(self):
```

Comment:
Why are these tests commented out?

Reply:
Ah, they should be deleted.  The methods they're testing don't exist now on VImageAdapter; they're just commented out from when I was first updating all the old tests.

The `test_downsample` below is commented out because it fails (on master too), because the test file isn't in the repo. Any idea what happened to it?

Reply:
Well the test seems to have been added [here](https://github.com/ecometrica/gdal2mbtiles/commit/539a490b66daa4c8e4daec78ca166f19d146534e), but it doesn't look like the test file was committed. You could make it by just clipping one of the other geotiffs to 258x258 with `gdal_translate`. Or I'm happy to do it if you want